addons/hr_holidays/models/hr_leave.py

```diff
@@ -839,7 +839,7 @@ def _prepare_holiday_values(self, employees):
             'request_date_from': self.date_from,
             'request_date_to': self.date_to,
             'notes': self.notes,
-            'number_of_days': work_days_data[employee.id]['days'],
+            'number_of_days': work_days_data[employee.id]['days'] if len(employees) > 1 else work_days_data['days'],
```

Comment:
Move this to 64374ea83c0d7b2745c8e440826591b3d6630bc3, and maybe split that to a version which always works in batch and a version which never does.